src/SYNFPAN.m

```diff
@@ -0,0 +1,379 @@
+ ;
+ i '$d(@troot) q 0  ;
+ s json=$na(@root@(ien,"json"))
+ ;
+ ; determine the patient
+ ;
+ n dfn
+ if $g(ien)'="" d  ;
+ . s dfn=$$ien2dfn^SYNFUTL(ien) ; look up dfn in the graph
+ else  d  ;
+ . s dfn=$g(args("dfn"))
+ . i dfn="" d  ;
+ . . n icn s icn=$g(args("icn"))
+ . . i icn'="" s dfn=$$icn2dfn^SYNFUTL(icn)
+ i $g(dfn)="" do  quit 0  ; need the patient
+ . s result("panels",1,"log",1)="Error, patient not found.. terminating"
+ ;
+ ;
+ new SYNZI s SYNZI=0
+ for  set SYNZI=$order(@troot@(SYNZI)) quit:+SYNZI=0  do  ;
+ . ;
+ . ; define a place to log the processing of this entry
+ . ;
+ . new jlog set jlog=$name(@eval@("panels",SYNZI))
+ . ;k @jlog
+ . ;
+ . ; ensure that the resourceType is DiagnosticReports
+ . ;
+ . new type set type=$get(@json@("entry",SYNZI,"resource","resourceType"))
+ . ;if type'="DiagnosticReport" do  quit  ;
+ . q:type'="DiagnosticReport"
+ . ;. ;set @eval@("panels",SYNZI,"vars","resourceType")=type
+ . ;. ;do log(jlog,"Resource type not DiagnosticReport, skipping entry")
+ . ;set @eval@("panels",SYNZI,"vars","resourceType")=type
+ . ;
+ . ; determine the DiagnosticReport category and quit if not a lab panel
+ . ;
+ . ;new obstype set obstype=$get(@json@("entry",SYNZI,"resource","category",1,"coding",1,"code"))
+ . new catcode set catcode=$get(@json@("entry",SYNZI,"resource","category",1,"coding",1,"code"))
+ . q:$$UP^XLFSTR(catcode)'["LAB"
+ . new loinc s loinc=""
+ . set loinc=$g(@json@("entry",SYNZI,"resource","code","coding",1,"code"))
+ . d log(jlog,"Panel loinc code is  "_loinc)
+ . ;
+ . ; see if this resource has already been loaded. if so, skip it
+ . ;
+ . if $g(ien)'="" if $$loadStatus("panels",SYNZI,ien)=1 do  quit  ;
+ . . d log(jlog,"Panel already loaded, skipping")
+ . ;
+ . ; determine Panel type, code, coding system, and display text
+ . ;
+ . new paneltype set paneltype=$get(@json@("entry",SYNZI,"resource","code","text"))
+ . if paneltype="" set paneltype=$get(@json@("entry",SYNZI,"resource","code","coding",1,"display"))
+ . do log(jlog,"Panel type is: "_paneltype)
+ . ;do log(jlog,"Panel type is: "_obsdisplay)
+ . set @eval@("panels",SYNZI,"vars","type")=paneltype
+ . ;
+ . ; determine the id of the resource
+ . ;
+ . new id set id=$get(@json@("entry",SYNZI,"resource","id"))
+ . set @eval@("panels",SYNZI,"vars","id")=id
+ . d log(jlog,"ID is: "_id)
+ . ;
+ . ;Here's the spec for uploading a panel:
+ . ;KBANTEST ;
+ . ;N %,SAM,RC
+ . ;S SAM("LAB_PANEL")="CHEM 7"
+ . ;S SAM("LAB_TEST","BUN")=10
+ . ;S SAM("LAB_TEST","CO2")=34
+ . ;S SAM("LAB_TEST","GLUCOSE")=180
+ . ;S SAM("PAT_SSN")="999504449"
+ . ;S SAM("RESULT_DT")="NOW"
+ . ;S SAM("LOCATION")="3E NORTH"
+ . ;S SAM("COLLECTION_SAMPLE")="BLOOD" ; optional
+ . ;S %=$$LAB^ISIIMP12(.RC,.SAM)
+ . ;ZWRITE RC
+ . ;QUIT
+ . ;
+ . ; starting the parameter array with the panel level elements
+ . ;
+ . n MISC ; parameter array
+ . ;
+ . ; lab panel
+ . ;
+ . N PANEL
+ . S PANEL=$$MAP^SYNQLDM(loinc,"vistapanel")
```

Comment:
I figured out why you don't show failed panels. You need to add a line here if a panel is not found.